src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -2228,6 +2228,11 @@ public static void Select(IList? checkRead, IList? checkWrite, IList? checkError
 
         private static int ToTimeoutMicroseconds(TimeSpan timeout)
         {
+            if (timeout == Timeout.InfiniteTimeSpan)
+            {
+                return -1;
+            }
```

Comment:
While this helps to make TimeSpan.InfiniteTimeSpan / TimeSpan.FromMilliseconds(-1) be treated as infinite, it feels weird to me that TimeSpan.FromMicroseconds(-1) is still also going to be treated as infinite.  We also need to decide if we care about values that are truncated to the same value, e.g. TimeSpan.FromMilliseconds(-1.1) is also going to be treated as infinite by other APIs that accept TimeSpans.

(And regardless of what we do here, the XML comments on the above method need to be updated, as they talk about -1 microseconds.)

Reply:
There is really no practical use for any negative numbers besides infinite, right? We can document that -1 (from Microseconds) and InfiniteTimeSpan are both treated equally. And we can reject rest of negative numbers IMHO.

If you have other APIs in mind for consistency please let me know.

Reply:
IMO, we should accept only `Timeout.InfiniteTimeSpan` from the negative numbers and converted in whatever value represents infinite in the underlying function. 

Also this method is called from `Poll`, so we should make sure the behavior for infinite is the same.

Finally, the xml comment of `Select` above should drop the detail about -1 microsecond as Stephen mentioned.

Reply:
updated + new test